presto-router/src/main/java/com/facebook/presto/router/predictor/PredictorManager.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.router.predictor;
+
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.airlift.json.JsonCodecFactory;
+import com.facebook.airlift.json.JsonObjectMapperProvider;
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.router.RouterConfig;
+import com.facebook.presto.router.spec.RouterSpec;
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import static com.fasterxml.jackson.databind.DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * The manager of fetching predicted resource usage of a SQL statement from the
+ * Presto query predictor.
+ * Note that it does not validate the SQL statements passed in.
+ */
+public class PredictorManager
+{
+    private static final Logger log = Logger.get(PredictorManager.class);
+    private static final JsonCodec<RouterSpec> CODEC = new JsonCodecFactory(
+            () -> new JsonObjectMapperProvider().get().enable(FAIL_ON_UNKNOWN_PROPERTIES))
+            .jsonCodec(RouterSpec.class);
+
+    private final RemoteQueryFactory remoteQueryFactory;
+    private final URI uri;
+
+    @Inject
+    public PredictorManager(RemoteQueryFactory remoteQueryFactory, RouterConfig config)
```

Comment:
this constructor looks little bit heavy to me,  can we move the logic to a separate initialization method?

Reply:
Nice call! I move the functionality to a `parseRouterConfig` method which returns an optional `RouterSpec` object.